src/client/components/form-elements/ErrorableTextInput.jsx

```diff
@@ -66,6 +67,10 @@ class ErrorableTextInput extends React.Component {
       requiredSpan = <span className="hca-required-span">*</span>;
     }
 
+    if (this.props.charMax) {
+      maxCharacters = (<small>(Max. {this.props.charMax} characters)</small>);
```

Comment:
I don't think it's a problem to have an error message say: "Please enter a valid name. Maximum 30 characters," or something like that. To me, it's more confusing to have validation feedback in 2 different formats: as an error message and as help text below the field. It's also unnecessary in the vast majority of cases to show this message, since most probably won't hit this max. I think it would be better to only show this message, then, if they hit the max.

I'm totally fine with using the html attribute instead of our js validation functions for this. I think it's more of a question of how your provide feedback on the fact that there is a maximum number of characters allowed. One way you could do this is:
```
if (this.props.errorMessage) {
    errorSpanId = `${this.inputId}-error-message`;
    errorSpan = <span className="usa-input-error-message" id={`${errorSpanId}`}>{this.props.errorMessage}</span>;
    inputErrorClass = 'usa-input-error';
    labelErrorClass = 'usa-input-error-label';
} else if (this.props.field.value.length > this.props.charMax) {
    errorSpanId = `${this.inputId}-error-message`;
    errorSpan = <span className="usa-input-error-message" id={`${errorSpanId}`}>Maximum 30 characters</span>;
    inputErrorClass = 'usa-input-error';
    labelErrorClass = 'usa-input-error-label';
}
```